opm/simulators/wells/StandardWell_impl.hpp

```diff
@@ -1892,15 +1893,17 @@ namespace Opm
         if (thp_controlled_well || bhp_controlled_well) {
 
             double total_rate = 0.0;
+            const double sign = this->isInjector() ? 1.0:-1.0;
             for (int phase = 0; phase < np; ++phase){
-                total_rate += ws.surface_rates[phase];
+                total_rate += sign * ws.surface_rates[phase];
             }
             // for pressure controlled wells the well rates are the potentials
             // if the rates are trivial we are most probably looking at the newly
             // opened well and we therefore make the affort of computing the potentials anyway.
-            if (std::abs(total_rate) > 0) {
+            if (total_rate > 0) {
+                const double sign = this->isInjector() ? 1.0:-1.0;
```

Comment:
The latest build check just alerted me to the fact that this `sign` object [shadows](https://ci.opm-project.org/job/opm-simulators-PR-builder/3326/gcc/fileName.1498975879/) an object with the same value in the outer scope.  You can probably just delete this object.